pkg/boot/initrd.go

```diff
@@ -21,6 +21,10 @@ func CatInitrds(initrds ...io.ReaderAt) io.ReaderAt {
 	}
 
 	return uio.NewLazyOpenerAt(strings.Join(names, ","), func() (io.ReaderAt, error) {
+		// TODO: be smarter
+		if len(initrds) == 1 {
```

Comment:
Why do it inside the lazy opener? I'd do it further up

Reply:
Yes, exactly, that needs some test cases to be updated :) I will do that

Reply:
If I move this up, i.e.,
```patch
diff --git a/pkg/boot/initrd.go b/pkg/boot/initrd.go
index 0acb49a1..fac240a0 100644
--- a/pkg/boot/initrd.go
+++ b/pkg/boot/initrd.go
@@ -15,16 +15,15 @@ import (
 // CatInitrds concatenates initrds on first ReadAt call from a list of
 // io.ReaderAts, pads them to a 512 byte boundary.
 func CatInitrds(initrds ...io.ReaderAt) io.ReaderAt {
+       if len(initrds) == 1 {
+               return initrds[0]
+       }
        var names []string
        for _, initrd := range initrds {
                names = append(names, stringer(initrd))
        }

        return uio.NewLazyOpenerAt(strings.Join(names, ","), func() (io.ReaderAt, error) {
-               // TODO: be smarter
-               if len(initrds) == 1 {
-                       return initrds[0], nil
-               }
                buf := new(bytes.Buffer)
                for i, ireader := range initrds {
                        size, err := buf.ReadFrom(uio.Reader(ireader))
```

I get:

```sh
go test ./pkg/boot/
--- FAIL: TestCatInitrds (0.00s)
    initrd_test.go:106: Cat([0xc00011e340]) = name &{yay %!s(int64=0) %!s(int=-1)}, want *strings.Reader
FAIL
FAIL	github.com/u-root/u-root/pkg/boot	0.003s
FAIL
```

And I have noooo clue :(